kura/org.eclipse.kura.core.cloud/src/main/java/org/eclipse/kura/core/cloud/CloudPayloadJsonDecoder.java

```diff
@@ -0,0 +1,170 @@
+/*******************************************************************************
+ * Copyright (c) 2017 Eurotech and/or its affiliates
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     Eurotech
+ *******************************************************************************/
+package org.eclipse.kura.core.cloud;
+
+import static org.eclipse.kura.core.cloud.CloudPayloadJsonFields.BODY;
+import static org.eclipse.kura.core.cloud.CloudPayloadJsonFields.POSITION;
+import static org.eclipse.kura.core.cloud.CloudPayloadJsonFields.TIMESTAMP;
+import static org.eclipse.kura.core.cloud.CloudPayloadJsonFields.CloudPayloadJsonPositionFields.ALTITUDE;
+import static org.eclipse.kura.core.cloud.CloudPayloadJsonFields.CloudPayloadJsonPositionFields.HEADING;
+import static org.eclipse.kura.core.cloud.CloudPayloadJsonFields.CloudPayloadJsonPositionFields.LATITUDE;
+import static org.eclipse.kura.core.cloud.CloudPayloadJsonFields.CloudPayloadJsonPositionFields.LONGITUDE;
+import static org.eclipse.kura.core.cloud.CloudPayloadJsonFields.CloudPayloadJsonPositionFields.PRECISION;
+import static org.eclipse.kura.core.cloud.CloudPayloadJsonFields.CloudPayloadJsonPositionFields.SATELLITES;
+import static org.eclipse.kura.core.cloud.CloudPayloadJsonFields.CloudPayloadJsonPositionFields.SPEED;
+import static org.eclipse.kura.core.cloud.CloudPayloadJsonFields.CloudPayloadJsonPositionFields.STATUS;
+import static org.eclipse.kura.core.cloud.CloudPayloadJsonTypes.BOOLEAN;
```

Comment:
Float cannot be added due to some database limitations (it will be stored as double).
The real problem now is that the types in the wires do not match the ones supported by the Kura Protobuf and this can cause exceptions while trying to publish.
The thing now to understand is if we want to adhere to the Kura Protobuf or not and how to deal in wires with unsupported types.